e2e/specs/translation-builder-workflows.spec.ts

```diff
@@ -0,0 +1,437 @@
+      const translationPanel = formBuilderPage.page
+        .getByRole('tabpanel')
+        .filter({ has: formBuilderPage.page.getByRole('button', { name: /upload translation/i }) });
+
+      await expect(translationPanel).toBeVisible();
+      await expect(translationPanel.getByRole('button', { name: /download translation/i })).toBeVisible();
+      await expect(translationPanel.getByRole('button', { name: /upload translation/i })).toBeVisible();
+      const tabList = translationPanel.getByRole('tablist', { name: 'Translation filter' });
+      await expect(tabList).toBeVisible();
+
+      const tabs = tabList.getByRole('tab');
+      await expect(tabs).toHaveCount(3);
+      await expect(tabs.nth(0)).toContainText(/all/i);
+      await expect(tabs.nth(1)).toContainText(/translated/i);
+      await expect(tabs.nth(2)).toContainText(/untranslated/i);
+      await expect(translationPanel.getByPlaceholder(/search translation keys/i)).toBeVisible();
+    });
+  });
+
+  test('Workflow 2: Language Selection and Switching', async ({ page }) => {
+    const formBuilderPage = new FormBuilderPage(page);
+
+    await test.step('When I visit the form builder', async () => {
+      await formBuilderPage.gotoFormBuilder();
+    });
+
+    await test.step('And I search for the form and open it for editing', async () => {
+      await formBuilderPage.searchForForm(formDetails.name);
+      await formBuilderPage.page.getByRole('row', { name: formDetails.name }).getByLabel('Edit Schema').first().click();
+    });
+
+    await test.step('And I open the Translation builder tab', async () => {
+      await formBuilderPage.translationBuilderTab().click();
+    });
+
+    await test.step('Then I should see the translation builder components (Download/Upload buttons, Translation filter tabs: All/Translated/Untranslated, and the search input)', async () => {
+      await expect(formBuilderPage.downloadTranslationButton()).toBeVisible();
+    });
+
+    await test.step('And the language dropdown is present and functional', async () => {
+      const languageDropdown = formBuilderPage.languageDropdown();
+      await expect(languageDropdown).toBeVisible();
+      await expect(languageDropdown).toBeEnabled();
+    });
+
+    await test.step('When I open the language dropdown and view available languages', async () => {
+      await formBuilderPage.languageDropdown().click();
+
+      const dropdownMenu = formBuilderPage.translationBuilderPanel().locator('.cds--list-box__menu');
+      await expect(dropdownMenu).toBeVisible();
+
+      const languageOptions = dropdownMenu.getByRole('option');
+      const optionCount = await languageOptions.count();
+      expect(optionCount).toBeGreaterThan(0);
+
+      await expect(dropdownMenu.getByRole('option', { name: 'English (en)' })).toBeVisible();
+      await expect(languageOptions.nth(1)).toBeVisible();
+    });
```

Comment:
Again, checking if visual elements are present are not steps in an e2e test. Things like these would get implicitly done in a step
```suggestion
    await test.step('And I open the Translation builder tab', async () => {
      await formBuilderPage.translationBuilderTab().click();
      await expect(formBuilderPage.downloadTranslationButton()).toBeVisible();
       const languageDropdown = formBuilderPage.languageDropdown();
      await expect(languageDropdown).toBeVisible();
      await expect(languageDropdown).toBeEnabled();
    });
```